ravem/indico_ravem/api.py

```diff
@@ -5,72 +5,84 @@
 
-    :returns: dict -- the status of the room as a JSON response according to the
-        RAVEM API.
-    """
-    room_name = room_name.replace('/', '-', 1)
-    return ravem_api_call('getstatus', method='GET', service_name='videoconference', where='room_name', value=room_name)
+        :param room_name: str -- the name of the physical room
 
+        :returns: dict -- the status of the room as a JSON response according to the
+            RAVEM API.
+        """
+        room_name = room_name.replace('/', '-', 1)
+        return ravem_api_call('rooms/details', method='GET',
+                              params={'where': 'room_name', 'value': room_name})
 
-def disconnect_endpoint(room_name, vc_room_name, service_type):
-    """Disconnects a physical room from a vidyo room using the RAVEM API.
+    @abstractmethod
+    def get_room_id(self, vc_room_data):
+        """Returns the provider specific room ID."""
+        pass
 
-    This call will return "OK" as a result immediately if RAVEM can (or has
-    started to) perform the operation. This does not mean the operation has
-    actually succeeded. One should poll for the status of the room afterwards
-    using the `get_endpoint_status` method after some delay to allow the
-    operation to be performed.
+    @abstractmethod
+    def connect_endpoint(self, room_name, vc_room_id):
+        """Connects a physical room to a videoconference room using the RAVEM API.
 
-    :param room_name: str -- the name of the physical room
-    :param vc_room_name: str -- The Vidyo room name (which is recommended to
-        disconnect the physical room from the Vidyo room in case the service
-        type is Vidyo.)
-    :param service_type: str -- The endpoint type (usually `vidyo` or `other`)
+        This call will return "OK" as a result immediately if RAVEM can (or has
+        started to) perform the operation. This does not mean the operation has
+        actually succeeded. One should poll for the status of the room afterwards
+        using the `get_endpoint_status` method after some delay to allow the
+        operation to be performed..
 
-    :returns: dict -- {'result': 'OK'} if the operation "succeeds", raises a
-        RavemAPIException otherwise.
-    """
-    room_name = room_name.replace('/', '-', 1)
-    params = {
-        'method': 'POST',
-        'where': 'room_name',
-        'value': room_name,
-        'type': service_type
-    }
-    if service_type == 'vidyo':
-        params['vidyo_room_name'] = vc_room_name
+        :param room_name: str -- the name of the physical room
+        :param vc_room_id: str -- the provider specific id of the
+            videoconference room
 
-    return ravem_api_call('videoconference/disconnect', **params)
+        :returns: dict -- {'result': 'OK'} if the operation "succeeds", raises a
+                  RavemAPIException otherwise.
+        """
+        pass
```

Comment:
not needed, the docstring already creates a valid function body

Reply:
I think this part was here before